File: Data Structures & Algorithms/sort-colors/submission-1.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        const int n = static_cast<int>(nums.size());
        int left = 0, right = n - 1;
        int i = 0;
        while (i <= right) {
            if (nums[i] == 0) {
                swap(nums[left], nums[i]);
                left ++;
                i++;
            } else if (nums[i] == 1) {
                i++;
            } else {
                swap(nums[i], nums[right]);
                right --;
            }
        }
    }
};

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-1.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        left, right = 0, 0
        n = len(nums)
        min_len = n + 1
        sum = 0

        while left < n:
            while right < n and sum < target and right - left <= min_len:
                sum += nums[right]
                right += 1

            if sum >= target and right - left + 1 <= min_len:
                min_len = min(min_len, right - left)

            sum -= nums[left]
            left += 1

        return 0 if min_len == n + 1 else min_len
